packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,545 @@
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'bork');
+      this.rerender();
+    });
+
+    this.assertText('BORK BORK');
+  }
+
+  ['@htmlbars should be able to render an unbound helper invocation']() {
+    this.registerHelper('-repeat', ([value], { count }) => {
+      let a = [];
+      while (a.length < count) {
+        a.push(value);
+      }
+      return a.join('');
+    });
+
+    this.render(`{{unbound (-repeat foo count=bar)}} {{-repeat foo count=bar}} {{unbound (-repeat foo count=2)}} {{-repeat foo count=4}}`, {
+      foo: 'X',
+      numRepeatsBinding: 'bar',
+      bar: 5
+    });
+
+    this.assertText('XXXXX XXXXX XX XXXX');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('XXXXX XXXXX XX XXXX');
+
+    this.runTask(() => set(this.context, 'bar', 1));
+
+    this.assertText('XXXXX X XX XXXX');
+
+    this.runTask(() => set(this.context, 'bar', 5));
+
+    this.assertText('XXXXX XXXXX XX XXXX');
+  }
+
+  ['@htmlbars should be able to render an bound helper invocation mixed with static values']() {
+    this.registerHelper('-surround', ([prefix, value, suffix]) => `${prefix}-${value}-${suffix}`);
+
+    this.render(`{{unbound (-surround prefix value "bar")}} {{-surround prefix value "bar"}} {{unbound (-surround "bar" value suffix)}} {{-surround "bar" value suffix}}`, {
+      prefix: 'before',
+      value: 'core',
+      suffix: 'after'
+    });
+
+    this.assertText('before-core-bar before-core-bar bar-core-after bar-core-after');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('before-core-bar before-core-bar bar-core-after bar-core-after');
+
+    this.runTask(() => {
+      setProperties(this.context, {
+        prefix: 'beforeChanged',
+        value: 'coreChanged',
+        suffix: 'afterChanged'
+      });
+    });
+
+    this.assertText('before-core-bar beforeChanged-coreChanged-bar bar-core-after bar-coreChanged-afterChanged');
+
+    this.runTask(() => {
+      setProperties(this.context, {
+        prefix: 'before',
+        value: 'core',
+        suffix: 'after'
+      });
+    });
+
+    this.assertText('before-core-bar before-core-bar bar-core-after bar-core-after');
+  }
+
+  ['@htmlbars should be able to render unbound forms of multi-arg helpers']() {
+    this.registerHelper('-fauxconcat', (params) => params.join(''));
+
+    this.render(`{{-fauxconcat foo bar bing}} {{unbound (-fauxconcat foo bar bing)}}`, {
+      foo: 'a',
+      bar: 'b',
+      bing: 'c'
+    });
+
+    this.assertText('abc abc');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('abc abc');
+
+    this.runTask(() => set(this.context, 'bar', 'X'));
+
+    this.assertText('aXc abc');
+
+    this.runTask(() => set(this.context, 'bar', 'b'));
+
+    this.assertText('abc abc');
+  }
+
+  ['@htmlbars should be able to render an unbound helper invocation for helpers with dependent keys']() {
```

Comment:
Port of https://github.com/emberjs/ember.js/blob/bf067f7370c3c3381352d68ef5dfe55d1178c6f7/packages/ember-htmlbars/tests/helpers/unbound_test.js#L384-L405